wasm/src/lsp/types.rs

```diff
@@ -491,3 +501,127 @@ fn test_doc_data_get_prev_position_1() {
     assert_eq!(want_line_jumps, have_line_jumps);
     assert_eq!(got_characters, all_expected_characters);
 }
+
+#[derive(Debug, Clone)]
+struct HelperWithDocRange {
+    pub loc: DocRange,
+}
+
+impl PartialEq for HelperWithDocRange {
+    fn eq(&self, other: &Self) -> bool {
+        self.loc == other.loc
+    }
+}
+
+impl PartialOrd for HelperWithDocRange {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.loc.cmp(&other.loc))
+    }
+}
+
+impl Eq for HelperWithDocRange {}
+impl Ord for HelperWithDocRange {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.loc.cmp(&other.loc)
+    }
+}
```

Comment:
Isn't this replaceable by `#[derive(PartialEq, PartialOrd, Eq, Ord)]` on `struct HelperWithDocRange`?

Reply:
Yes, thanks.  I had planned for that to contain more stuff than it did and missed that it no longer needed broken out comparisons.